sed/core/logging.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module provides a function to set up logging for the application. It configures
+both console and file logging handlers, allowing different log levels for each. The
+log files are stored in a user-specific log directory.
+
+"""
+import logging
+
+from sed.core.user_dirs import USER_LOG_PATH
+
+
+def setup_logging(name: str, verbose: bool = False, debug: bool = False) -> logging.Logger:
+    """
+    Configures and returns a logger with specified log levels for console and file handlers.
+
+    Args:
+        name (str): The name of the logger.
+        verbose (bool): If True, sets the console log level to INFO. Defaults to False.
+        debug (bool): If True, sets the console log level to DEBUG. Defaults to False.
+
+    Returns:
+        logging.Logger: The configured logger instance.
+
+    The logger will always write DEBUG level messages to a file located in the user's log
+    directory, while the console log level can be adjusted based on the 'verbose' and 'debug'
+    flags.
+    """
+    # Create logger
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)  # Set the minimum log level for the logger
+
+    # Determine console log level
+    if debug:
+        console_level = logging.DEBUG
+    elif verbose:
+        console_level = logging.INFO
+    else:
+        console_level = logging.WARNING
+
+    # Create console handler and set level to warning (default)
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(console_level)
+
+    # Create formatter for console
+    console_formatter = logging.Formatter("%(message)s")
+    console_handler.setFormatter(console_formatter)
+
+    # Add console handler to logger
+    logger.addHandler(console_handler)
+
+    log_file = USER_LOG_PATH.joinpath("sed.log")
```

Comment:
Shouldn't the file names contain e.g. a date or so?

Reply:
Currently everything is logged into one file but your idea makes sense. Should be an easy addition. I can either update it here, split to another PR or merge this PR and update logging afterwards in new PR.

Reply:
Apart from splitting by date, I would also strongly suggest to keep the log files together with the notebooks. If you put them into a global log file, it's going to be impossible to assign which lines come from which notebook, run etc.